client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -279,6 +406,10 @@ function mediaButton( editor ) {
 			// autosave take effect
 			editor.fire( 'change' );
 		}
+
+		if ( numOfImagesToUpdate === 0 ) {
```

Comment:
Maybe keep a list of dirty media ids, and update this out of the image update loop.